Samples/StorageProviders/StorageProviders/BaseJSONStorageProvider.cs

```diff
@@ -134,9 +142,16 @@ protected static string ConvertToStorageFormat(IGrainState grainState)
         /// <param name="entityData">JSON storage format representaiton of the grain state.</param>
         protected static void ConvertFromStorageFormat(IGrainState grainState, string entityData)
         {
-            JavaScriptSerializer deserializer = new JavaScriptSerializer();
-            object data = deserializer.Deserialize(entityData, grainState.State.GetType());
-            grainState.State = data;
+            JsonConvert.PopulateObject(entityData, grainState.State);
+        }
+    }
+
+    public class OrleansStateContractResolver : DefaultContractResolver
+    {
+        protected override IList<JsonProperty> CreateProperties(Type type, MemberSerialization memberSerialization)
+        {
+            var properties = base.CreateProperties(type, memberSerialization);
+            return properties.Where(p => p.PropertyName != "Etag").ToList();
```

Comment:
@jdom I just ran the sample and confirmed that Etag is not used. Removing it now.